pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -69,7 +69,7 @@ func NewCodecForScheme(
 	encodeVersion []unversioned.GroupVersion,
 	decodeVersion []unversioned.GroupVersion,
 ) runtime.Codec {
-	return NewCodec(encoder, decoder, scheme, scheme, scheme, runtime.ObjectTyperToTyper(scheme), encodeVersion, decodeVersion)
+	return NewCodec(encoder, decoder, runtime.UnsafeObjectConvertor(scheme), scheme, scheme, runtime.ObjectTyperToTyper(scheme), encodeVersion, decodeVersion)
```

Comment:
So are we assuming this safe by default since during a conversion step every sane thing I can think of throws away the original or is there some reason I'm not seeing that makes this always safe?